webpack.config.js

```diff
@@ -1,56 +1,77 @@
     path.resolve(__dirname, 'app/stylesheets/main.scss'),
   ],
   output: {
     path: path.resolve(__dirname, 'dist'),
     publicPath: '/',
-    filename: './dist.js'
+    filename: './dist.js',
   },
   module: {
     rules: [
-      { test: /\.css$/, include: path.resolve(__dirname, 'app'), loader: 'style-loader!css-loader' },
       {
-        test: /\.scss$/,
+        test: /\.s?css$/,
+        include: path.resolve(__dirname, 'app'),
+        loader: [MiniCssExtractPlugin.loader, 'css-loader', 'sass-loader'],
+      },
+      {
+        test: /\.ts(x?)$/,
+        exclude: /node_modules/,
+        use: [
+          {
+            loader: 'ts-loader',
+          },
+        ],
+      },
+      // All output '.js' files will have any sourcemaps re-processed by 'source-map-loader'.
+      {
+        enforce: 'pre',
+        test: /\.js$/,
         exclude: /node_modules/,
-        use: ExtractTextPlugin.extract({
-          fallback: 'style-loader',
-          use: [
-            'css-loader',
-            { loader: 'sass-loader', query: { sourceMap: false } },
-          ],
-          publicPath: '../'
-        }),
+        loader: 'source-map-loader',
       },
-      { test: /\.js[x]?$/, include: [
-        path.resolve(__dirname, 'app'),
-        path.resolve(__dirname, 'node_modules/sn-components-api/dist/dist.js')
-      ], exclude: /node_modules/, loader: 'babel-loader' }
-    ]
+    ],
+  },
+  externals: {
+    'filesafe-js': 'filesafe-js',
   },
   resolve: {
-    extensions: ['.js', '.jsx', '.css', '.scss'],
+    modules: ['node_modules', path.resolve(__dirname, 'app')],
+    extensions: ['.ts', '.tsx', '.js', '.jsx', '.css', '.scss'],
     alias: {
-        highlightjs_css: path.join(__dirname, 'node_modules/highlight.js/styles/atom-one-light.css'),
-        stylekit: path.join(__dirname, 'node_modules/sn-stylekit/dist/stylekit.css')
-    }
+      highlightjs_css: path.join(
+        __dirname,
+        'node_modules/highlight.js/styles/atom-one-light.css'
+      ),
+      stylekit: path.join(
+        __dirname,
+        'node_modules/sn-stylekit/dist/stylekit.css'
+      ),
+    },
   },
   plugins: [
-    new ExtractTextPlugin({ filename: './dist.css', disable: false, allChunks: true}),
+    new MiniCssExtractPlugin({
+      // Options similar to the same options in webpackOptions.output
+      // both options are optional
+      filename: 'dist.css',
+    }),
     new webpack.DefinePlugin({
-      'process.env': {
-        NODE_ENV: JSON.stringify('production')
-      }
+      'process.env.NODE_ENV': JSON.stringify('production'),
     }),
-    new CopyWebpackPlugin([
-      { from: './app/index.html', to: 'index.html' },
-    ])
-  ]
+    new CopyWebpackPlugin({
+      patterns: [
+        { from: './app/index.html', to: 'index.html' },
+        // This is for hosting KaTeX, which we currently don't do:
+        //{from: path.resolve(__dirname, './node_modules/katex/dist'), to: 'katex'},
```

Comment:
I don't get this? Why have this comment if it's not something we do?

Reply:
I thought we might need it later and we could save ourselves future work by leaving the comment there. I'll remove it